src/Features/LanguageServer/Protocol/Handler/CodeActions/RunCodeActionsHandler.cs

```diff
@@ -0,0 +1,85 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.Commands;
+using Newtonsoft.Json.Linq;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+{
+    /// <summary>
+    /// Runs a code action as a command on the server.
+    /// This is done when a code action cannot be applied as a WorkspaceEdit on the LSP client.
+    /// For example, all non-ApplyChangesOperations must be applied as a command.
+    /// TO-DO: Currently, any ApplyChangesOperation that adds or modifies an outside document must also be
+    /// applied as a command due to an LSP bug (see https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/).
+    /// Commands must be applied from the UI thread in VS.
+    /// </summary>
+    [ExportExecuteWorkspaceCommand(CodeActionsHandler.RunCodeActionCommandName)]
+    internal class RunCodeActionsHandler : IExecuteWorkspaceCommandHandler
+    {
+        private readonly ICodeFixService _codeFixService;
+        private readonly ICodeRefactoringService _codeRefactoringService;
+        private readonly ILspSolutionProvider _solutionProvider;
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public RunCodeActionsHandler(
+            ICodeFixService codeFixService,
+            ICodeRefactoringService codeRefactoringService,
+            ILspSolutionProvider solutionProvider,
+            IThreadingContext threadingContext)
+        {
+            _codeFixService = codeFixService;
+            _codeRefactoringService = codeRefactoringService;
+            _solutionProvider = solutionProvider;
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<object> HandleRequestAsync(
+            LSP.ExecuteCommandParams request,
+            LSP.ClientCapabilities clientCapabilities,
+            CancellationToken cancellationToken)
+        {
+            var runRequest = ((JToken)request.Arguments.Single()).ToObject<CodeActionResolveData>();
+            var document = _solutionProvider.GetDocument(runRequest.TextDocument);
+            var codeActions = await CodeActionHelpers.GetCodeActionsAsync(document, _codeFixService, _codeRefactoringService,
+                runRequest.Range, cancellationToken).ConfigureAwait(false);
```

Comment:
```suggestion
            var codeActions = await CodeActionHelpers.GetCodeActionsAsync(
                document, _codeFixService, _codeRefactoringService, runRequest.Range, cancellationToken).ConfigureAwait(false);
```

stylistic nit for consideration.  both of these split the line to two lines.  however, this version keeps teh parameters together, instead of splitting at a random point.